xonsh/environ.py

```diff
@@ -687,10 +687,14 @@ def _is_executable_file(path):
 
 
 def yield_executables_windows(directory, name):
+    casefold_name = name.casefold()
     extensions = builtins.__xonsh_env__.get('PATHEXT')
     for a_file in os.listdir(directory):
         base_name, ext = os.path.splitext(a_file)
-        if name == base_name and ext.upper() in extensions:
+
+        if (
+            casefold_name == base_name.casefold() or casefold_name == a_file.casefold()
+        ) and ext.upper() in extensions:
```

Comment:
Yes. It was on windows. But I must admit that I don't have any experience with such files names. It seems like windows interpret them as different files. 
![image](https://cloud.githubusercontent.com/assets/1038978/14061984/82d9f476-f38f-11e5-8e7e-e03347c93c81.png)